packages/cli-kit/src/output.ts

```diff
@@ -22,8 +23,10 @@ import {AbortController, AbortSignal} from 'abort-controller'
 import cjs from 'color-json'
 import stripAnsi from 'strip-ansi'
 import {Writable} from 'node:stream'
+import {WriteStream, createWriteStream} from 'node:fs'
```

Comment:
@gonzaloriestra @pepicrft thanks for the review! I took your comments to heart and honestly, it's true, it's not a great API as originally designed.

I decided to try the more performant `fs.createWriteStream` which lets us stream things out asynchronously without creating a new file descriptor each time.

I'm not sure about the official word on reliability, but I do know that in local testing, it seemed to output errors correctly, which means that the stream is always cleared before program exit, solving the original pain point.

This change allowed me to clean up a lot of the other noise I'd introduced. So let me know what you think!